app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transfers/TransfersScreenWrapper.kt

```diff
@@ -132,20 +144,71 @@ private fun DetailPane(
     navigator: ThreePaneScaffoldNavigator<DestinationContent>,
     hasTransfer: () -> Boolean,
 ) {
-
     val destinationContent = navigator.safeCurrentContent()
 
     if (destinationContent == null) {
         NoSelectionEmptyState(hasTransfer())
+    } else if (destinationContent.folderUuid != null) {
+        val windowAdaptiveInfo = LocalWindowAdaptiveInfo.current
+        FilesDetailsScreen(
+            windowAdaptiveInfo,
+            navigator,
+            destinationContent.folderUuid,
+            destinationContent.direction,
+            destinationContent.transferUuid,
+            navigateToFolder = { selectedFolderUuid ->
+                navigator.navigateToFolder(
+                    destinationContent.direction,
+                    destinationContent.transferUuid,
+                    selectedFolderUuid,
+                )
+            }
+        )
     } else {
         TransferDetailsScreen(
             transferUuid = destinationContent.transferUuid,
             direction = destinationContent.direction,
             navigateBack = ScreenWrapperUtils.getBackNavigation(navigator),
+            navigateToFolder = { selectedFolderUuid ->
+                navigator.navigateToFolder(
+                    destinationContent.direction,
+                    destinationContent.transferUuid,
+                    selectedFolderUuid,
+                )
+            },
         )
     }
 }
 
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+@Composable
+private fun FilesDetailsScreen(
+    windowAdaptiveInfo: WindowAdaptiveInfo,
+    navigator: ThreePaneScaffoldNavigator<DestinationContent>,
+    folderUuid: String,
+    transferDirection: TransferDirection,
+    transferUuid: String,
+    navigateToFolder: (folderUuid: String) -> Unit,
+) {
+    FilesDetailsScreen(
+        navigateToFolder = { selectedFolderUuid ->
+            navigateToFolder(selectedFolderUuid)
+        },
+        folderUuid = folderUuid,
+        navigateBack = { navigator.popBackStack() },
+        close = {
+            if (windowAdaptiveInfo.isWindowSmall()) navigator.navigateBack()
+            navigator.navigateToDetails(
+                windowAdaptiveInfo,
+                transferDirection,
+                transferUuid
+            )
+        },
```

Comment:
I'm not sure to get what this does exactly but, does it not leave the whole FilesDetailsScreen flow altogether no matter how deep in folders you are when you click on the close button?